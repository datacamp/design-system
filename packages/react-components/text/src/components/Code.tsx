import tokens from '@datacamp/waffles-tokens';
import { computeDataAttributes } from '@datacamp/waffles-utils';
import { css } from '@emotion/react';

import codeStyle from '../codeStyle';

export interface CodeProps {
  /**
   * The code to display.
   */
  children: string;
  /**
   * Sets the css class of the rendered element. Can be used to apply custom
   * styles.
   */
  className?: string;
  /**
   * As with all the other waffles components, dataAttributes can be used to set
   * data- html attributes on the element.
   */
  dataAttributes?: { [key: string]: string };
}

const style = css(codeStyle, {
  backgroundColor: tokens.colors.beige,
  margin: '0 2px',
  padding: '2px 4px',
  whiteSpace: 'nowrap',
});

const Code = ({
  children,
  className,
  dataAttributes,
}: CodeProps): JSX.Element => {
  const parsedDataAttributes = computeDataAttributes(dataAttributes);

  return (
    <code className={className} css={style} {...parsedDataAttributes}>
      {children}
    </code>
  );
};

export default Code;
