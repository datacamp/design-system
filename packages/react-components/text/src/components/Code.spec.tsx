import '@testing-library/jest-dom/extend-expect';

import axeRender from '@datacamp/waffles-axe-render';
import React from 'react';

import Code from './Code';

describe('<Code /', () => {
  it('renders the text in a <code> element', async () => {
    const exampleText = 'this is some code';
    const { getByText } = await axeRender(<Code>{exampleText}</Code>);
    expect((getByText(exampleText) as HTMLElement).tagName).toEqual('CODE');
  });
});

it('sets the className on the code element', async () => {
  const className = 'test-class';
  const { container } = await axeRender(
    <Code className={className}>example code</Code>
  );

  expect(container.firstChild).toHaveClass(className);
});

it('sets the dataAttributes on the code element', async () => {
  const data = { test: 'example' };
  const { container } = await axeRender(
    <Code dataAttributes={data}>example code</Code>
  );

  expect(container.firstChild).toHaveAttribute('data-test', 'example');
});

describe('snapshots', () => {
  it('renders', async () => {
    const { container } = await axeRender(<Code>example code</Code>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
