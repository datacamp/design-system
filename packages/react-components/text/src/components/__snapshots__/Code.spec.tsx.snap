// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders 1`] = `
.emotion-0 {
  border: 0;
  border-radius: 4px;
  color: rgb(61,66,81);
  font-family: monospace;
  font-family: JetBrainsMonoNL;
  font-size: 14px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  background-color: rgb(235,244,247);
  box-shadow: 0 0 0 1px #e6eaeb;
  margin-left: 2px;
  margin-right: 4px;
  padding-bottom: 0;
  padding-left: 2px;
  padding-right: 2px;
  padding-top: 0;
}

<code
  class="emotion-0"
>
  example code
</code>
`;
